packages/suite/src/hooks/suite/useCoinjoinTracker.ts

```diff
@@ -0,0 +1,63 @@
+import { onCancel as closeModal, openModal } from '@suite-actions/modalActions';
+import { CoinjoinSession, RoundPhase } from '@suite-common/wallet-types';
+import { TrezorDevice } from '@suite-types/index';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useInterval } from 'react-use';
+
+const CRITICAL_PHASES = [
+    RoundPhase.ConnectionConfirmation,
+    RoundPhase.OutputRegistration,
+    RoundPhase.TransactionSigning,
+];
+
+export const useCoinjoinTracker = (
+    accountKey: string,
+    session: CoinjoinSession,
+    relatedDevice: TrezorDevice | undefined,
+) => {
+    const [isCriticalPhase, setIsCriticalPhase] = useState(false);
+    const [isDeviceLost, setIsDeviceLost] = useState(false);
+
+    const dispatch = useDispatch();
+
+    useInterval(() => {
+        const { phase } = session;
```

Comment:
Will be a part of https://github.com/trezor/trezor-suite/blob/1be74a2f30b65f4c0aa1567e7e6208f78410256b/packages/suite/src/actions/wallet/coinjoinClientActions.ts#L135